uxarray/plot/dataarray_plot.py

```diff
@@ -61,3 +71,159 @@ def datashade(uxda: UxDataArray,
         _cmap = cmap
 
     return tf.shade(aggregated, cmap=_cmap, **kwargs)
+
+
+def _point_raster(uxda: UxDataArray,
+                  backend: Optional[str] = "bokeh",
+                  pixel_ratio: Optional[float] = 1.0,
+                  dynamic: Optional[bool] = False,
+                  precompute: Optional[bool] = True,
+                  projection: Optional[ccrs] = None,
+                  width: Optional[int] = 1000,
+                  height: Optional[int] = 500,
+                  colorbar: Optional[bool] = True,
+                  cmap: Optional[str] = "Blues",
+                  aggregator: Optional[str] = "mean",
+                  interpolation: Optional[str] = "linear",
+                  npartitions: Optional[int] = 1,
+                  **kwargs):
+
+    if uxda.face_centered():
+        # data mapped to face centroid coordinates
+        data_mapping = "center"
+        lon = uxda.uxgrid.Mesh2_face_x.values
+        lat = uxda.uxgrid.Mesh2_face_y.values
+    elif uxda.node_centered():
+        # data mapped to face corner coordinates
+        data_mapping = "corner"
+        lon = uxda.uxgrid.Mesh2_node_x.values
+        lat = uxda.uxgrid.Mesh2_node_y.values
+    else:
+        raise ValueError(
+            "Issue with data. It is neither face-centered nor node-centered!")
+
+    data_values = uxda.values
+
+    recompute = True
+    if data_mapping == "center":
+        if uxda.uxgrid._centroid_points_df_proj[
+                0] is not None and uxda.uxgrid._centroid_points_df_proj[
+                    1] == projection:
+            recompute = False
+            points_df = uxda.uxgrid._centroid_points_df_proj[0]
+
+    else:
+        if uxda.uxgrid._corner_points_df_proj[
+                0] is not None and uxda.uxgrid._corner_points_df_proj[
+                    1] == projection:
+            recompute = False
+            points_df = uxda.uxgrid._corner_points_df_proj[0]
+
+    if recompute:
+        # need to recompute points & projection
+        if projection is not None:
+            lon, lat, _ = projection.transform_points(ccrs.PlateCarree(), lon,
```

Comment:
What is the cost of calculating this again and again vs. storing this? e.g. Time and memory costs, respectively, for the 3.75km maybe? **For point rasterization**, execution costs for any of the functions being called in this method were reasonable small as far as I remember, so any caching might be unnecessary.

Reply:
These timings are a rough benchmark, however this is what I observed.

![image](https://github.com/UXARRAY/uxarray/assets/67855069/d45c36fb-41b2-4f54-b5b5-cef7d4b5daa5)

Additionally, the caching here is of the Dask dataframe, not just the transformed coordinates.

Reply:
Yeah, that's the Dask dataframe. This chart shows great the exceution times. What's the memory space our cached attributes take?

Reply:
For the 3.75km dataset, the Dask dataframe occupies about 3 * (42mb) for the (lon, lat, data var) when persisted in memory. This is in addition to the list that stores the data frame, along with the projection reference, which are essentially points, so it is minimal .

Reply:
Is it bit (mb) or byte (MB)? If bit, that seems pretty viable. If byte though, 3 * 42 MB may still seem reasonable but adds to the total memory usage UXarray has (along with the other cached data). I've seen memory usages in the margins of 32 GB while performing such visualizations with 7.5km data, which seems huge. I couldn't say anything further for now since we don't have any detailed analysis, but we should take that into account.

One intermediary solution maybe the `cache=False` optional arg, and if user wants it `True` we would do caching (as we did that with other caching things in the past).

Reply:
This was in Mega Bytes

I will add a cache parameter to the function.

Reply:
Oops, if it is MB, then yeah having an optional arg makes it a bit safer I think.

Reply:
Good suggestion, I've added it.